internal/ast/utilities.go

```diff
@@ -3633,3 +3637,79 @@ func GetSemanticJsxChildren(children []*JsxChild) []*JsxChild {
 		}
 	})
 }
+
+func IsExpandoPropertyDeclaration(node *Node) bool {
+	if node == nil {
+		return false
+	}
+	return IsPropertyAccessExpression(node) || IsElementAccessExpression(node) || IsBinaryExpression(node)
+}
+
+func GetExpandoInitializer(initializer *Node, isPrototypeAssignment bool) *Node {
+	if initializer.Kind == KindCallExpression {
+		expr := SkipParentheses(initializer.Expression())
+		if expr.Kind == KindFunctionExpression || expr.Kind == KindArrowFunction {
+			return initializer
+		}
+		return nil
+	}
+
+	if initializer.Kind == KindFunctionExpression || initializer.Kind == KindCallExpression || initializer.Kind == KindArrowFunction {
+		return initializer
+	}
+
+	if initializer.Kind == KindObjectLiteralExpression && (len(initializer.Properties()) == 0 || isPrototypeAssignment) {
+		return initializer
+	}
+
+	return nil
+}
+
+func GetEffectiveInitializer(node *Node) *Expression {
+	if IsInJSFile(node) && node.Initializer() != nil && IsBinaryExpression(node.Initializer()) {
+		initializer := node.Initializer().AsBinaryExpression()
+		if initializer.OperatorToken.Kind == KindBarBarToken || initializer.OperatorToken.Kind == KindQuestionQuestionToken {
+			if node.Name() != nil && IsEntityNameExpressionEx(node.Name(), IsInJSFile(node)) && IsSameEntityName(node.Name(), initializer.Left) {
+				return initializer.Right
+			}
+		}
+	}
+	return node.Initializer()
+}
+
+func GetDeclaredExpandoInitializer(node *Node) *Expression {
```

Comment:
In particular, this suddenly needing to be implemented is suspect. It used to be used all over in the checker in the old expando checking paths in checker, but now it isn't at all and was only added for the declaration emit - it implies we're doing something wrong and ignoring the new parser output for expandos.